sql/descriptor.go

```diff
@@ -62,12 +62,15 @@ type descriptorProto interface {
 }
 
 // checkPrivilege verifies that p.user has `privilege` on `descriptor`.
-func (p *planner) checkPrivilege(descriptor descriptorProto, privilege privilege.Kind) error {
-	if descriptor.GetPrivileges().CheckPrivilege(p.user, privilege) {
+func (p *planner) checkPrivilege(descriptor descriptorProto, priv privilege.Kind) error {
+	if descriptor.GetPrivileges().CheckPrivilege(p.user, priv) {
 		return nil
 	}
+	if priv != privilege.SELECT && isSystemConfigID(descriptor.GetID()) {
+		return fmt.Errorf("system database is read-only")
```

Comment:
except it really isn't, some objects (DB `system` and tables `namespace` and `descriptor`) are indeed read-only from the SQL p.o.v, but other system tables allow `INSERT`, `DELETE`, and `UPDATE`.

The hard-coded sets of privileges on those tables are the absolute requirement for `root` (eg: no more and no less than those privileges), and are the maximum sets of privileges allowed for all other users. Still doesn't make things read-only though, just special.